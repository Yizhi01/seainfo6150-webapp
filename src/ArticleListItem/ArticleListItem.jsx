import React, { useState } from "react";
import PropTypes from "prop-types";
import ArticleTextToggleButton from "../AtricleTextToggleButton/ArticleTextToggleButton";
import styles from "./ArticleListItem.module.css";

const ArticleListItem = (props) => {
    const [text, setText] = useState("Show more");
    function onToggleButton(e, text) {
        e.preventDefault();
        if (text === "Show less") {
            setText("Show more");
        } else {
            setText("Show less");
        }
    }
    let displayContent;
    if (text === "Show less") {
        displayContent = (
            <article>
                <h2>{props.article.title}</h2>
                <p>{props.article.shortText}</p>
                <time dateTime={props.article.timeStamp}>{props.article.displayDate} </time>
            </article>
        )
    } else {
        displayContent = <h2>{props.article.title}</h2>;
    }

    return (
        <li className={styles.article_item}>
            {displayContent}
            <ArticleTextToggleButton buttonText={text} onClick={onToggleButton} />
        </li>
    );
};

ArticleListItem.propTypes = {
    article: PropTypes.object.isRequired,
};

export default ArticleListItem;