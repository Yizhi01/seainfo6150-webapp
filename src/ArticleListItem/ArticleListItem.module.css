.article_item {
    margin-top: 30px;
    padding: 20px;
    border: 1px solid red;
    border-radius: 10px;
    list-style: none;
    background-image: linear-gradient(to right, rgb(205, 92, 92), white 15%);
    box-shadow: 5px 10px 20px grey;
}

.article_item h2 {
    margin: auto 0;
    color: rgb(220, 20, 60);
    font-size: 36px;
}

.article_item time {
    font-weight: bold;
}