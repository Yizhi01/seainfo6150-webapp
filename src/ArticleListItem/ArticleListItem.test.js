import React from "react";
import ArticleListItem from "./ArticleListItem";
import { fireEvent } from "@testing-library/react";

describe("ArticleListItem tests", () => {
  it("renders correctly after button is clicked", () => {
    const testArtile = {'title' : 'Test'};
    const { container, getByText } = render(
      <ArticleListItem article={testArtile} />
    );
    const button = getByText("Show more");
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });
});