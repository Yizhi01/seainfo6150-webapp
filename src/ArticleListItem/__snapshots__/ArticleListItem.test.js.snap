// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly after button is clicked 1`] = `
<div>
  <li
    class="container"
  >
    <article
      class="article"
    >
      <div
        class="wrapper"
      >
        <h2
          class="title"
        >
          Test
        </h2>
        <div
          class="text"
        >
          <p />
          <time
            class="time"
          />
        </div>
      </div>
      <button
        class="button"
      >
        Show less
      </button>
    </article>
  </li>
</div>
`;
